api/src/main/java/org/openmrs/module/pihcore/setup/MetadataSharingSetup.java

```diff
@@ -0,0 +1,150 @@
+package org.openmrs.module.pihcore.setup;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.metadatasharing.ImportConfig;
+import org.openmrs.module.metadatasharing.ImportMode;
+import org.openmrs.module.metadatasharing.ImportedPackage;
+import org.openmrs.module.metadatasharing.MetadataSharing;
+import org.openmrs.module.metadatasharing.api.MetadataSharingService;
+import org.openmrs.module.metadatasharing.wrapper.PackageImporter;
+import org.openmrs.module.pihcore.PihCoreUtil;
+
+import java.io.BufferedInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.Iterator;
+import java.util.List;
+
+public class MetadataSharingSetup {
+
+    protected static final Log log = LogFactory.getLog(MetadataSharingSetup.class);
+
+    public static void installMetadataSharingPackages() {
+
+        try {
+            Collection<File> mdsFiles = loadMdsFiles();
+
+            if (mdsFiles == null) {
+                log.error("Unable to find any metadata sharing files");
+                return;
+            }
+
+            List<PackageImporter> packageImporters = loadPackageImporters(mdsFiles);
+            removeAlreadyLoadedPackageImporters(packageImporters);
+            sortPackageImporters(packageImporters);
+            loadPackages(packageImporters);
```

Comment:
All this logic seems quite a lot more complicated than what it replaces, which is

```
MetadataUtil.setupSpecificMetadata(getClass().getClassLoader(), "some", "package", "names")
```

Why aren't we doing it that way, with emrapi?

Reply:
Yeah, I had been hoping/assuming we'd just us the EmrApi methods, but I ran into a few hiccups.  First, the EmrApi functionality was based around loading the packages from the classpath, but, more, importantly, the EmrApi method depends on the "packages.xml" external file to determine what version a package represents and what the groupUuid of that package is.

The new way loads all the files it finds in the concept directory and then retrieves the groupUuid  and the version number by parsing the package itself. (I though this might be time-consuming, but it turned out not to be).

So I was annoyingly unable to use the EmrApi code directly.  Since this is likely a shorter term fix I just copied and reworked code as necessary from EmrApi into PIH Core rather than updating EmrApi to support these changes (it's also likely that no one other that us uses these EmrApi uitl methods).